### PR TITLE
feat: get change-log api

### PR DESCRIPTION
**Pull request by Mryashbhardwaj**
depends on https://github.com/goto/proton/pull/104

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -32,7 +32,8 @@ scheduler-resource-test:
 generate-proto: ## regenerate protos
 	@echo " > generating protobuf from goto/proton"
 	@echo " > [info] make sure correct version of dependencies are installed using 'make install'"
-	@buf generate https://github.com/goto/proton/archive/${PROTON_COMMIT}.zip#strip_components=1 --template buf.gen.yaml --path gotocompany/optimus
+	# @buf generate https://github.com/goto/proton/archive/${PROTON_COMMIT}.zip#strip_components=1 --template buf.gen.yaml --path gotocompany/optimus
```
Comment (sbchaos) [inline Makefile:35]: Can we revert this change

File: core/job/handler/v1beta1/job.go [hunk 2]
```diff
@@ -49,6 +49,7 @@ type JobService interface {
 	ChangeNamespace(ctx context.Context, jobSourceTenant, jobNewTenant tenant.Tenant, jobName job.Name) error
 	Delete(ctx context.Context, jobTenant tenant.Tenant, jobName job.Name, cleanFlag, forceFlag bool) (affectedDownstream []job.FullName, err error)
 	Get(ctx context.Context, jobTenant tenant.Tenant, jobName job.Name) (jobSpec *job.Job, err error)
+	GetChangelog(ctx context.Context, projectName tenant.ProjectName, jobName job.Name) ([]*job.ChangeLog, error)
```
Comment (sbchaos) [inline core/job/handler/v1beta1/job.go:52]: Can we break this service or create a new service for this. ?
This interface is very bloated.

Comment (Mryashbhardwaj) [reply core/job/handler/v1beta1/job.go:52]: sure

File: Makefile (hunk 1 above)
Comment (Mryashbhardwaj) [reply Makefile]: done sir

**Review by sbchaos — APPROVED** [review]
